stl/inc/xstring

```diff
@@ -2873,13 +2851,7 @@ public:
         auto&& _Alproxy         = _GET_PROXY_ALLOCATOR(_Alty, _Getal());
         _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data); // throws
 
-#if _HAS_CXX20
-        const bool _Activate_large_mode = _New_capacity < _New_size || _STD is_constant_evaluated();
-#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
-        const bool _Activate_large_mode = _New_capacity < _New_size;
-#endif // _HAS_CXX20
-
-        if (_Activate_large_mode) {
+        if (_New_capacity < _New_size) {
             // we should never allocate less than _BUF_SIZE space (_New_size could be small if constant evaluated)
             const size_type _Requested_size = (_STD max)(_New_size, _BUF_SIZE);
             _New_capacity                   = _Calculate_growth(_Requested_size, _BUF_SIZE - 1, max_size());

```

Comment:
This is TRANSITION logic that should be updated.

Above, we initialized `size_type _New_capacity = _BUF_SIZE - 1;` and it hasn't changed yet. The condition `_New_capacity < _New_size` means that:

* `_BUF_SIZE - 1 < _New_size`
* `_New_size > _BUF_SIZE - 1`
* `_New_size >= _BUF_SIZE`

Therefore, `(_STD max)(_New_size, _BUF_SIZE)` is always `_New_size`.

We should drop this comment, drop `_Requested_size`, and directly use `_New_size`.